model/test_results.go

```diff
@@ -88,6 +89,51 @@ func (t *TestResults) Find(ctx context.Context) error {
 	return nil
 }
 
+// TestResultsFindOptions allows for querying with or without execution val
+type TestResultsFindOptions struct {
+	TaskID         string
+	Execution      int
+	EmptyExecution bool
+}
+
+// FindByTaskExecution searches the database for the TestResults associated with the provided task_id / execution pair.
+// The enviromemt should not be nil. If execution is empty, will default to most recent execution
+func (t *TestResults) FindByTaskID(ctx context.Context, opts TestResultsFindOptions) error {
+	if t.env == nil {
+		return errors.New("cannot find with a nil environment")
+	}
+
+	if opts.TaskID == "" {
+		return errors.New("cannot find without a task_id")
+	}
+
+	t.populated = false
+	findOneOpts := options.FindOne().SetSort(bson.D{{Key: bsonutil.GetDottedKeyName(testResultsInfoKey, testResultsInfoExecutionKey), Value: -1}})
+	err := t.env.GetDB().Collection(testResultsCollection).FindOne(ctx, createTestResultsFindQuery(opts), findOneOpts).Decode(t)
+	if db.ResultsNotFound(err) {
+		if opts.EmptyExecution {
+			return errors.Wrapf(err, "could not find test results record with task_id %s in the database", opts.TaskID)
+		} else {
```

Comment:
You don't need the `else` if you return in the `if` block - you can just `return errors.Wrapf(err, "could not find test results record...")`. Usually, if an `if` ends in `return`, in Go, you typically don't use an `else` clause after it since the `else` is already implied.